pkg/linstorhelper/client.go

```diff
@@ -18,11 +19,26 @@ import (
 	"github.com/piraeusdatastore/piraeus-operator/v2/pkg/vars"
 )
 
+// optsCache stores the common LINSTOR Client options for all clients. Every reconcile run may create a new client, but
+// they should all share the same Options such as caches and rate limits, provided they point at the same controller.
+var optsCache sync.Map
+
 // Client is a LINSTOR client with convenience functions.
 type Client struct {
 	lapi.Client
 }
 
+// PerClusterOption makes an option that is shared between all instances connecting to the same cluster.
+// This can be used to make a cached shared between different client instances or a rate limiter applied to all
+// clients connected to the same controller.
+func PerClusterOption(opt func() lapi.Option) lapi.Option {
```

Comment:
This is a function which returns a function and has an argument that is a function that returns a function 😵